### ISSUE TITLE
File glob appears to work incorrectly with and

### ISSUE THREAD

**Issue by Wideshanks** [xonsh/xonsh#1871]
I want to use `and` with two commands and use file globs in both, but ~~it appears to be re-using the same glob in the second command~~ it's not getting both globs correctly. What am I missing here?

``` bash
$ ls -l
Actually_test.txt
Actually.txt
Complete_test.txt
Complete.txt

$ echo *.txt and echo *_test.txt # expect second command to give only two files
Actually.txt Actually_test.txt Complete.txt Complete_test.txt
Actually.txt Actually_test.txt Complete.txt Complete_test.txt

$ echo *_test.txt # these are the two files I wanted
Actually_test.txt Complete_test.txt

$ echo *_test.txt and echo *.txt # same result if reversed
Actually.txt Actually_test.txt Complete.txt Complete_test.txt
Actually.txt Actually_test.txt Complete.txt Complete_test.txt

$ bash
$ echo *.txt && echo *_test.txt # bash gives me the files I wanted in the second command
Actually_test.txt Actually.txt Complete_test.txt Complete.txt
Actually_test.txt Complete_test.txt

$ echo *_test.txt && echo *.txt
Actually_test.txt Complete_test.txt
Actually_test.txt Actually.txt Complete_test.txt Complete.txt

$ exit
$ xonfig
+------------------+-----------------+
| xonsh            | 0.4.5           |
| Git SHA          | None            |
| Python           | 3.4.0           |
| PLY              | 3.8             |
| have readline    | True            |
| prompt toolkit   | 1.0.7           |
| shell type       | prompt_toolkit  |
| pygments         | 2.0.2           |
| on posix         | True            |
| on linux         | True            |
| distro           | Ubuntu          |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+
```

**Comment by gforsyth** [comment]
Hey @Wideshanks -- thanks for reporting!  Just to note that the current xonsh release is 0.4.7, but I can reproduce this on current master. 

And it's not reusing the first glob, so much as not parsing the second one correctly, it seems:

```console
gil@bad_cat ~/tmp $ ls
bar  foo  one.txt  one_test.txt  two.txt  two_test.txt
gil@bad_cat ~/tmp $ echo *.txt and echo *_test.txt
one.txt one_test.txt two.txt two_test.txt
bar foo one.txt one_test.txt two.txt two_test.txt
```

but if the `*` isn't the leading character then it seems to behave itself:

```console
gil@bad_cat ~/tmp $ echo *.txt and echo two*.txt
one.txt one_test.txt two.txt two_test.txt
two.txt two_test.txt
```

**Comment by gforsyth** [comment]
Yep, looks like a parser issue.

```console
gil@bad_cat ~/tmp $ echo *.txt and echo two*.txt
<xonsh-code>:1:6:12 - echo *.txt and echo two*.txt
<xonsh-code>:1:6:12 + ![echo *.txt] and echo two*.txt                  
<xonsh-code>:1:23 - ![echo *.txt] and echo two*.txt
<xonsh-code>:1:23 + ![echo *.txt] and ![echo two*.txt]
```
vs.
```console
gil@bad_cat ~/tmp $ echo *.txt and echo *_test.txt
<xonsh-code>:1:6:12 - echo *.txt and echo *_test.txt
<xonsh-code>:1:6:12 + ![echo *.txt] and echo *_test.txt
<xonsh-code>:1:6:12 - echo *.txt and echo *_test.txt
<xonsh-code>:1:6:12 + ![echo *.txt] and echo *_test.txt
```

**Comment by scopatz** [comment]
Potential fix in #2273

**Comment by scopatz** [comment]
The fix here has been merged, please feel free to reopen if needed!
